PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:7 + add:0, change matches:
-    _inherit = ['purchase.order']
-    _inherit = ['res.company']
-    _inherit = ["stock.move"]
-    _inherit = ['stock.warehouse.orderpoint']
-    _inherit = ['stock.picking']
-    _inherit = ['stock.rule']
-    _inherit = ['stock.warehouse']

Total Changes: 16

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/mrp_subcontracting_dropshipping/models/purchase.py
+++ b/addons/mrp_subcontracting_dropshipping/models/purchase.py
@@ -6,7 +6,7 @@ from odoo.exceptions import UserError
 
 
 class PurchaseOrder(models.Model):
-    _inherit = ['purchase.order']
+    _inherit = 'purchase.order'
 
     default_location_dest_id_is_subcontracting_loc = fields.Boolean(related='picking_type_id.default_location_dest_id.is_subcontracting_location')
 

--- a/addons/mrp_subcontracting_dropshipping/models/res_company.py
+++ b/addons/mrp_subcontracting_dropshipping/models/res_company.py
@@ -5,7 +5,7 @@ from odoo import api, fields, models
 
 
 class ResCompany(models.Model):
-    _inherit = ['res.company']
+    _inherit = 'res.company'
 
     dropship_subcontractor_pick_type_id = fields.Many2one('stock.picking.type')
 

--- a/addons/mrp_subcontracting_dropshipping/models/stock_move.py
+++ b/addons/mrp_subcontracting_dropshipping/models/stock_move.py
@@ -5,7 +5,7 @@ from odoo import models
 
 
 class StockMove(models.Model):
-    _inherit = ["stock.move"]
+    _inherit = "stock.move"
 
     def _prepare_procurement_values(self):
         vals = super()._prepare_procurement_values()

--- a/addons/mrp_subcontracting_dropshipping/models/stock_orderpoint.py
+++ b/addons/mrp_subcontracting_dropshipping/models/stock_orderpoint.py
@@ -5,7 +5,7 @@ from odoo import models
 
 
 class StockWarehouseOrderpoint(models.Model):
-    _inherit = ['stock.warehouse.orderpoint']
+    _inherit = 'stock.warehouse.orderpoint'
 
     def _prepare_procurement_values(self, date=False, group=False):
         vals = super()._prepare_procurement_values(date, group)

--- a/addons/mrp_subcontracting_dropshipping/models/stock_picking.py
+++ b/addons/mrp_subcontracting_dropshipping/models/stock_picking.py
@@ -6,7 +6,7 @@ from odoo.tools import float_compare
 
 
 class StockPicking(models.Model):
-    _inherit = ['stock.picking']
+    _inherit = 'stock.picking'
 
     def _compute_is_dropship(self):
         dropship_subcontract_pickings = self.filtered(lambda p: p.location_dest_id.is_subcontracting_location and p.location_id.usage == 'supplier')

--- a/addons/mrp_subcontracting_dropshipping/models/stock_replenish_mixin.py
+++ b/addons/mrp_subcontracting_dropshipping/models/stock_replenish_mixin.py
@@ -5,7 +5,7 @@ from odoo.osv import expression
 
 
 class StockReplenishMixin(models.AbstractModel):
-    _inherit = ['stock.replenish.mixin']
+    _inherit = 'stock.replenish.mixin'
 
     def _get_allowed_route_domain(self):
         domains = super()._get_allowed_route_domain()

--- a/addons/mrp_subcontracting_dropshipping/models/stock_rule.py
+++ b/addons/mrp_subcontracting_dropshipping/models/stock_rule.py
@@ -5,7 +5,7 @@ from odoo import models
 
 
 class StockRule(models.Model):
-    _inherit = ['stock.rule']
+    _inherit = 'stock.rule'
 
     def _prepare_purchase_order(self, company_id, origins, values):
         if 'partner_id' not in values[0] \

--- a/addons/mrp_subcontracting_dropshipping/models/stock_warehouse.py
+++ b/addons/mrp_subcontracting_dropshipping/models/stock_warehouse.py
@@ -5,7 +5,7 @@ from odoo import api, fields, models, _
 
 
 class StockWarehouse(models.Model):
-    _inherit = ['stock.warehouse']
+    _inherit = 'stock.warehouse'
 
     subcontracting_dropshipping_to_resupply = fields.Boolean(
         'Dropship Subcontractors', default=True,
